homeassistant/components/starlink/binary_sensor.py

```diff
@@ -0,0 +1,136 @@
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import StarlinkData, StarlinkUpdateCoordinator
+from .entity import StarlinkEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up all binary sensors for this entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        StarlinkBinarySensorEntity(coordinator, description)
+        for description in BINARY_SENSORS
+    )
+
+
+@dataclass
+class StarlinkBinarySensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[StarlinkData], bool | None]
+
+
+@dataclass
+class StarlinkBinarySensorEntityDescription(
+    BinarySensorEntityDescription, StarlinkBinarySensorEntityDescriptionMixin
+):
+    """Describes a Starlink binary sensor entity."""
+
+
+class StarlinkBinarySensorEntity(StarlinkEntity, BinarySensorEntity):
+    """A BinarySensorEntity for Starlink devices. Handles creating unique IDs."""
+
+    entity_description: StarlinkBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: StarlinkUpdateCoordinator,
+        description: StarlinkBinarySensorEntityDescription,
+    ) -> None:
+        """Initialize the binary sensor."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{self.coordinator.data.status['id']}_{description.key}"
+
+    @property
+    def is_on(self) -> bool | None:
+        """Calculate the binary sensor value from the entity description."""
+        return self.entity_description.value_fn(self.coordinator.data)
+
+
+BINARY_SENSORS = [
+    StarlinkBinarySensorEntityDescription(
+        key="roaming",
+        name="Roaming mode",
+        value_fn=lambda data: data.alert["alert_roaming"],
+    ),
+    StarlinkBinarySensorEntityDescription(
+        key="currently_obstructed",
+        name="Obstructed",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        value_fn=lambda data: data.status["currently_obstructed"],
+    ),
+    StarlinkBinarySensorEntityDescription(
+        key="heating",
+        name="Heating",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.alert["alert_is_heating"],
+    ),
+    StarlinkBinarySensorEntityDescription(
+        key="power_save_idle",
+        name="Idle",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.alert["alert_is_power_save_idle"],
+    ),
+    StarlinkBinarySensorEntityDescription(
+        key="install_pending",
+        name="Update pending",
+        device_class=BinarySensorDeviceClass.UPDATE,
```

Comment:
On means update available, Off means up-to-date. The use of this device class should be avoided, please consider using the [update](https://developers.home-assistant.io/docs/core/entity/update) entity instead.

Reply:
I will look into the update platform next. Buttons (and switches) are relatively small, would it make sense to tack update on the end of the same PR?

Reply:
Better to do one platform per PR.  Its a lot easier to find small blocks of time to do reviews.  Its the big PRs that sit for a long time because they need large chunks of time

Reply:
On `update` platform - how should I handle this situation? Starlink doesn't expose the "latest version" as far as we know, it just says "there is an update"

Reply:
We talked about this and it looks like that ends up falling into the exception case since there is no good solution currently when you don't know the new version.

So good to add back in a new PR.  Sorry the churn.